Bjean/interest helper method

benjamin2jean opened the pull request:
In MvcController.java
- I added functionality to applyInterest(). This method will check that a user valid number of desposits ($20+) has reached 5 or is about to reach 5 before applying interest to the user's balance. 

-Adjusted the structure of applyInterest() to work so that when a user first logins or when currently doing a deposit.

benjamin2jean committed:
added interest helper method to de-clutter controller logic

benjamin2jean committed:
Added functionality to the applyInterest() in MvcController

s1ddh4rthc commented:
Good job with Part 1. Make sure you follow codebase conventions in your function and make your commit messages better. Make sure you finish the integ tests for part 2.

benjamin2jean committed:
added integration tests to ensure applyInterest() works as intended also cleaned and refactored applyInterest()

benjamin2jean commented:
In MvcController.java
-I cleaned up and refactored the code of applyInterest() to be more inline with clean code style
-Added comments to explain the flow of the program
-The deposit used in applyInterest() has been adjusted to use more precision to prevent rounding errors that were possible before. e.g.$19.99 were valid before but no longer is
-Added a new TransactionHistory string to help indicating that interest was applied to a customer's balance

In MvcControllerIntegTest.java
-Added Integrity Test to test possible interactions with applyInterest() such as below $20 deposit,above $20 deposit, a $19.99 deposit, a $20.01 deposit, etc.
-These tests also check that the TransactionHistory are updated when applicable

Current Status of Tests:Passing

benjamin2jean committed:
Added Integration Tests for Cryptocurrency Solana

benjamin2jean commented:
In MvcControllerIntegTest.java
-Added comments to explain the flow of the program

-Added an Integrity Test that tries to buy ETC, buy SOL, sell SOL is currently failing
Purpose: To ensure that a user can buy multiple different cryptocurrencies and sell a potion of the cryptocurrencies
Current Status of Test(s):0/1 Failing
Reason:It doesn't seem that the Tester and Non-Tester objects work well together when multiple are used.
TestName: testCryptoBuyETH_BuySOL_SellSOL

-Added Integrity Test(s) to test possible interactions such as buying and selling unsupported cryptocurrencies e.g. Bitcoin.
Purpose: Unsupported cryptocurrencies should not be processed and return the user to the "welcome" screen
Current Status of Test(s): 2/2 Pass
TestName(s): testCryptoBuyBTCInvalidCase, testCryptoSellBTCInvalidCase

Note: Currently waiting for T.A.'s advice on how to address testCryptoBuyETH_BuySOL_SellSOL

Note for T.A.s: I also tried using the Tester Object to represent a user and the non-Tester Object(s) as the test(s) but I run into an AbstractMap error

benjamin2jean committed:
Added my Final Project idea
